leanix_admin/tag_group.py

```diff
@@ -0,0 +1,153 @@
+import json
+
+import leanix_admin.file as file
+import leanix_admin.graphql as graphql
+from leanix_admin.action import BackupAction, RestoreAction
+
+
+def find_by_name(needle, haystack):
+    for current in haystack:
+        if needle['name'] == current['name']:
+            return current
+    return None
+
+
+class TagGroupsBase:
+    def __init__(self, http, graphql_url):
+        self.http = http
+        self.graphql_url = graphql_url
+
+    def _fetch_tag_groups(self, erase_id=True):
+        response = self._exec_graphql(graphql.list_tag_groups)
+        tag_groups = []
+        for tag_group_edge in response.get('listTagGroups', {}).get('edges', []):
+            tag_group = tag_group_edge['node']
+            tags = []
+            for tag_edge in tag_group.get('tags', {}).get('edges', []):
+                tag = tag_edge['node']
+                if erase_id:
+                    del tag['id']
+                tags.append(tag)
+
+            tag_group['tags'] = tags
+            if erase_id:
+                del tag_group['id']
+            tag_groups.append(tag_group)
+        return tag_groups
+
+    def _exec_graphql(self, query, variables=None):
+        if variables is None:
+            variables = {}
+        body = {'operationName': None,
+                'query': query,
+                'variables': variables}
+        r = self.http.post(self.graphql_url, json=body)
+        r.raise_for_status()
+        r_body = r.json()
+        errors = r_body.get('errors', None)
+        if errors:
+            print(errors)
+            print('Request: ', body)
+            raise Exception()
+        data = r_body.get('data', None)
+        if not data:
+            print('Request: ', body)
+            raise Exception('Empty response data')
+        return data
+
+
+class TagGroupsBackupAction(TagGroupsBase, BackupAction):
+    def __init__(self, http, graphql_url):
+        TagGroupsBase.__init__(self, http, graphql_url)
+        BackupAction.__init__(self, name='tag-groups')
+        self.http = http
+        self.graphql_url = graphql_url
+
+    def do_perform(self):
+        tag_groups = self._fetch_tag_groups()
+        file.write_to_disk(self.name, tag_groups)
+
+
+class TagGroupsRestoreAction(TagGroupsBase, RestoreAction):
+    def __init__(self, http, graphql_url):
+        TagGroupsBase.__init__(self, http, graphql_url)
+        RestoreAction.__init__(self, name='tag-groups')
+        self.http = http
+        self.graphql_url = graphql_url
+
+    def do_perform(self):
```

Comment:
This is another function where I as an outsider would appreciate a little explanation...